frontend/catalyst/compiler.py

```diff
@@ -310,6 +312,9 @@ class LLVMIRToObjectFile(PassPipeline):
     _default_flags = [
         "--filetype=obj",
         "--relocation-model=pic",
+        # -O0 is used to achieve compile times similar to -regalloc=fast and disabling
+        # -twoaddrinst. However, from the command line, one cannot disable -twoaddrinst
```

Comment:
A question: is it possible to disable `-twoaddrinst` using the C++ interface? What do you think, can we later revert to the default opt level if apply both settings from the compilation driver which is currently in development?

Reply:
I don't know if it is possible to disable `-twoaddrinst` from the C++ interface. I suspect there is a way.

I want to note here that `-twoaddrinst` is taking too long when `-O2` is used, however `-twoaddrinst` is also used in `-O0` but takes a significantly lower amount of time. It is likely that another pass transforms the IR in such a way that enables `-twoaddrinst` to be more powerful, but I was not fully able to track it.

I think what would be important in the long term is to study a little bit the literature to find out which register allocator strategies are used when JIT compiling (is Linear Scan Register Allocator still preferred or not?) and try to generate code which avoids this compile time explosion during register allocation (use smaller functions as opposed to large functions with many variables). Maybe at some point we may consider writing our own register allocator (but that is farther away). One question that I haven't answered is what register allocation strategy is used by the LLVM JIT library.

Reply:
I'll mark this comment as resolved in the absence of further discussion 👍